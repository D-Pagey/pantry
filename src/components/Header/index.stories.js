import React from 'react';
import { storiesOf } from '@storybook/react';
import { Header } from '.';

storiesOf('Header', module).add('default', () => <Header />).add('on add page', () => <Header page="Add an item"/>);
