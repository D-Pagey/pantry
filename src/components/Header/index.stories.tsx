import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import { UnreadNotification, UserDan, WelcomeNotification } from '../../fixtures';
import { AuthContext } from '../ProviderAuth';
import { Header } from '.';

export default { title: 'Header' };

export const normal = () => (
    <MemoryRouter>
        <AuthContext.Provider
            value={{
                fetchUserData: () => {},
                isAuthed: true,
                isCheckingAuth: false,
                setUser: () => {},
                signOut: () => {},
                user: { ...UserDan, notifications: [WelcomeNotification, UnreadNotification] }
            }}
        >
            <Header />
        </AuthContext.Provider>
    </MemoryRouter>
);

// move to knobs

// export const withPageProps = () => (
//     <MemoryRouter>
//         <Header page="Add an item" />
//     </MemoryRouter>
// );
