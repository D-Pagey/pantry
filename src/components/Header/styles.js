import styled from 'styled-components/macro';
import { NavLink } from 'react-router-dom';

export const Header = styled.header`
    display: flex;
    justify-content: space-between;
    margin: 0 0 2rem;
    padding: 2rem 0 0;
`;

export const Link = styled(NavLink)`
    color: initial;
    text-decoration: none;
`;

export const Title = styled.h4`
    font-weight: 500;
    margin: 0;
`;

export const List = styled.ul`
    list-style: none;
    padding: 0;
    margin: 0;
    display: flex;
`;

export const Item = styled.li`
    margin: 0 0 0 2rem;

    &:first-child {
        margin: 0;
    }
`;

export const StyledNavLink = styled(NavLink)`
    text-decoration: none;
    color: grey;

    &:hover {
        color: black;
    }
`;
