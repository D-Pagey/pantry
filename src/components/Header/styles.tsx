import styled from 'styled-components/macro';
import { NavLink } from 'react-router-dom';

export const Wrapper = styled.div`
    align-items: center;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,.06);
    display: flex;
    padding: 33px 24px;
    left: 0; 
    position: fixed;
    right: 0;
    top: 0;
`;

export const Link = styled(NavLink)`
    color: initial;
    text-decoration: none;
`;

export const Title = styled.h4`
    font-size: 25px;
    font-weight: 500;
    margin: 0;
    padding: 0 0 0 14px;
`;

export const Logo = styled.img`
    position: relative;
    top: 2px;
    width: 32px;
`;

export const Arrow = styled.img`
    cursor: pointer;
    height: 16px;
    width: 16px;
`;