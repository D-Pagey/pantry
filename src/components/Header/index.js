import React from 'react';
import * as S from './styles';

const Header = () => (
    <header>
        <S.Title>TroubleMaker</S.Title>
    </header>
);

export default Header;
