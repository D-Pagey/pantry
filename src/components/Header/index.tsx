import React from 'react';
import Media from 'react-media';
import * as S from './styles';

const Header: React.FC = () => (
    <S.Header>
        <S.Link to="/">
            <S.Title>Pantry</S.Title>
        </S.Link>

        <Media
            query="(min-width: 476px)"
            render={(): JSX.Element => (
                <S.List data-testid="desktopNavList">
                    <S.Item>
                        <S.StyledNavLink to="/" exact>Check Fridge</S.StyledNavLink>
                    </S.Item>
                    <S.Item>
                        <S.StyledNavLink to="/add">Add Food</S.StyledNavLink>
                    </S.Item>
                </S.List>
            )}
        />
    </S.Header>
);

export default Header;
