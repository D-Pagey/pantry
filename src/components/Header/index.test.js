import React from 'react';
import Header from '.';

describe('Header component', () => {
    it('should render', () => {
        const { container } = render(<Header />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render desktop nav on large devices', () => {
        setMatchMedia(1000);
        const { getByTestId } = render(<Header />);
        getByTestId('desktopNavList');
    });
});
