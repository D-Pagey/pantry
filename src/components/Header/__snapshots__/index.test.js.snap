// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 0 2rem;
  padding: 2rem 0 0;
}

.c1 {
  color: initial;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  font-weight: 500;
  margin: 0;
}

<header
  class="c0"
>
  <a
    aria-current="page"
    class="c1 active"
    href="/"
  >
    <h4
      class="c2"
    >
      Pantry
    </h4>
  </a>
</header>
`;
